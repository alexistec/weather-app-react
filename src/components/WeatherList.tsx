import React from 'react';
import {Card,CardContent,Avatar} from '@material-ui/core';
import moment from 'moment';


type WeatherListProps = {
  days : object[]
} 

const WeatherList:React.FC<WeatherListProps> = ({days}) => {
    return(
        <div style={{display: 'flex',justifyContent:'center',alignItems:'center'}}>
            {days.map((value:any,index:number) =>( 
                <Card style={{width:180,marginLeft:15,justifyContent:'center',alignItems:'center'}} key={index}>
                  <CardContent>
                    <h3>{moment(value.dt_txt).format('dddd')}</h3>
                    <Avatar style={{height:50,width:50}}alt="Remy Sharp" src={`http://openweathermap.org/img/wn/${value.weather[0].icon}.png`}  />
                    <h3>{value.main.temp}Â°</h3>
                  </CardContent>
                </Card>
            ))}
        </div>
    )
    
}

export default WeatherList;