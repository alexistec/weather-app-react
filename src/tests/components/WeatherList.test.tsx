import React from "react";
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import WeatherList from "../../components/WeatherList";



describe('Test in Component <WeatherList/>', () => {
    const days = [
        {
            "dt": 1630519200,
            "main": {
                "temp": 33.82,
                "feels_like": 32.19,
                "temp_min": 33.82,
                "temp_max": 35.49,
                "pressure": 1012,
                "sea_level": 1012,
                "grnd_level": 1002,
                "humidity": 24,
                "temp_kf": -1.67
            },
            "weather": [
                {
                    "id": 800,
                    "main": "Clear",
                    "description": "clear sky",
                    "icon": "01d"
                }
            ],
            "clouds": {
                "all": 2
            },
            "wind": {
                "speed": 4.35,
                "deg": 22,
                "gust": 5.3
            },
            "visibility": 10000,
            "pop": 0,
            "sys": {
                "pod": "d"
            },
            "dt_txt": "2021-09-01 18:00:00"
        },
        {
            "dt": 1630540800,
            "main": {
                "temp": 25.34,
                "feels_like": 24.77,
                "temp_min": 25.34,
                "temp_max": 25.34,
                "pressure": 1009,
                "sea_level": 1009,
                "grnd_level": 1002,
                "humidity": 32,
                "temp_kf": 0
            },
            "weather": [
                {
                    "id": 801,
                    "main": "Clouds",
                    "description": "few clouds",
                    "icon": "02n"
                }
            ],
            "clouds": {
                "all": 17
            },
            "wind": {
                "speed": 3.22,
                "deg": 58,
                "gust": 11.93
            },
            "visibility": 10000,
            "pop": 0,
            "sys": {
                "pod": "n"
            },
            "dt_txt": "2021-09-02 00:00:00"
        }        
    ];

    test('should render correctly', () => {
        const wrapper = renderer.create(<WeatherList days={ days }/>)
                        .toJSON();
        expect( wrapper ).toMatchSnapshot();
    })
    
})
