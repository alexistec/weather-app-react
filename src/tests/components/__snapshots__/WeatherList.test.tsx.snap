// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Component <WeatherList/> should render correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginLeft": 15,
        "width": 180,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h3>
        Wednesday
      </h3>
      <div
        className="MuiAvatar-root MuiAvatar-circular"
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      >
        <img
          alt="Remy Sharp"
          className="MuiAvatar-img"
          src="http://openweathermap.org/img/wn/01d.png"
        />
      </div>
      <h3>
        33.82
        °
      </h3>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginLeft": 15,
        "width": 180,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h3>
        Thursday
      </h3>
      <div
        className="MuiAvatar-root MuiAvatar-circular"
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      >
        <img
          alt="Remy Sharp"
          className="MuiAvatar-img"
          src="http://openweathermap.org/img/wn/02n.png"
        />
      </div>
      <h3>
        25.34
        °
      </h3>
    </div>
  </div>
</div>
`;
